src/frontend/contexts/AuthContext.tsx

```diff
@@ -32,9 +33,19 @@ export const AuthProvider = ({children}: {children: React.ReactNode}) => {
   const [authState, setAuthState] = React.useState<AuthState>(
     passcode === null ? 'authenticated' : 'unauthenticated',
   );
-
+  const isE2E = process.env.EXPO_PUBLIC_E2E_TEST === 'true';
   const shareDialogIsOpen = useIsShareDialogOpen();
 
+  React.useEffect(() => {
+    if (!isE2E) {
+      if (passcode !== null) {
+        FlagSecureModule.activate();
+      } else {
+        FlagSecureModule.deactivate();
+      }
+    }
+  }, [passcode, isE2E]);
+
```

Comment:
I think this would be clearer and catch any cases where flagsecure was activated during e2e if you don't nest the statement:

```ts

if(passcode !== null && !isE2E){
   FlagSecureModule.activate();
} else{
    // now if flag secure was activated somewhere else but we are in e2e, then it will be deactivated
    FlagSecureModule.deactivate();
}

```